### PR TITLE
Make remote get/set of answers work in UI

### PR DESCRIPTION
**Pull request by stephenjust**
I had to make pretty significant changes to the activities to make this work.
Due to all dataManager operations being necessarily asynchronous, and not working at all without a callback, it became cleaner to trigger creation of answer records in the question activity through a callback. To do so in the add answer fragment would have required several more levels of nested callbacks, and then to duplicate them again in the question activity.

On the bright side, besides being a lot of code, it does look kind of nice.

### CODE REVIEW

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/QuestionActivity.java [hunk 1]
```diff
@@ -159,17 +162,44 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-	// method for the add answer button, to given question
-    public void addAnswer(View view){
-    	FragmentManager fm = getFragmentManager();
-    	Bundle QuesBox = new Bundle();
-    	QuesBox.putString( "Qid", question.getId().toString());
-    	AddAnswerDialogFragment aA = new AddAnswerDialogFragment();
-    	aA.setArguments(QuesBox);
-    	aA.show(fm, "addanswerdialogfragmentlayout");
-    }
-
```
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/QuestionActivity.java:172]: I don't understand this deletion/insertion... how are you putting the Qid into the AddAnswerDialogFragment?

Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/QuestionActivity.java:172]: Basically, with the QuestionActivity responsible for creating the answer, the answer dialog fragment doesn't need to know what the question is. Which nicely avoids a bunch of annoying callbacks in the fragment, which I have no nice way of handling errors from.

**Comment by stephenjust** [comment]
@Kopo101 Review?

**Comment by Cotto4** [comment]
Checked it out, I'll have to trust you guys that it works. Most of it looks to be either ES stuff (which I'll defer to you) or Callback based stuff. Looking at the Question and Answer activities, the callback method looks fine to me. I don't know the intricacies of how they work, but the theory seems to be solid. If you want me to look at anything in detail just let me know, but from a general overview it looks positively orgasmic.

**Pull request merged**